earth_enterprise/src/common/performancelogger.cpp

```diff
@@ -38,22 +40,32 @@ void PerformanceLogger::log(
     const size_t size) {
 
   const timespec duration = timespecDiff(endTime, startTime);
-  const pid_t tid = syscall(SYS_gettid);
   stringstream message;
 
   message.setf(ios_base::fixed, ios_base::floatfield);
-  message << setprecision(9)
-          << operation << " " << object << ": "
-          << "start time: " << startTime
-          << ", "
-          << "end time: " << endTime
-          << ", "
-          << "duration: " << duration
-          << ", "
-          << "thread: " << tid;
-  if (size > 0) {
-    message << ", size: " << size;
-  }
+  message << startTime << ", "
+          << endTime << ", "
+          << duration << ", "
+          << size << ", "
+          << operation << ", "
+          << object;
 
   notify(NFY_NOTICE, "%s\n", message.str().c_str());
 }
+
+// Thread safety wrapper for log output
+void PerformanceLogger::do_notify( const string& message, ostream& out, khMutex& mutex ) {
+
+  // Get the thread ID
+  pthread_t tid = pthread_self();
+  pid_t pid = getppid();  // get the ID of the parent process
```

Comment:
I think what we want here is `getpid`, not `getppid`, though having both wouldn't hurt.